repl/src/test/scala/ammonite/repl/ScriptTests.scala

```diff
@@ -7,92 +10,300 @@ object ScriptTests extends TestSuite{
-          error: Compilation Failed
-          Main.scala:\\d\\+: not found: value res
-          res
-          ^
+            @ val r = res
+            r: String = ${"\"\"\""}
+            <a href="www.google.com">omg</a>
+            ${"\"\"\""}
+            """)
+        }
+        'preserveImports{
+          val typeString =
+            if (!scala.util.Properties.versionString.contains("2.10"))
+              """Left[String, Nothing]"""
+            else
+              """util.Left[String,Nothing]"""
+          check.session(s"""
+            @ load.module("$scriptPath/PreserveImports.scala")
 
-          """)
+            @ val r = res
+            r: $typeString = Left("asd")
+            """)
+        }
+        'annotation{
+          check.session(s"""
+            @ load.module("$scriptPath/Annotation.scala")
+
+            @ val r = res
+            r: Int = 24
+            """)
+        }
+        'syntax{
+          check.session(s"""
+            @ load.module("$scriptPath/BlockSepSyntax.scala")
+
+            @ val r = res
+            r: Int = 24
+            """)
+        }
       }
-      'nofile{
-        check.session(s"""
-          @ load.script("$scriptPath/notHere")
-          error: java.nio.file.NoSuchFileException: $scriptPath/notHere
-          """
-        )
+      'failures{
+        'syntaxError{
+          check.session(s"""
+            @ load.exec("$scriptPath/SyntaxError.scala")
+
+            @ val r = res
+            error: Compilation Failed
+            Main.scala:\\d\\+: not found: value res
+            res
+            ^
+            """)
+        }
+        'compilationError{
+          check.session(s"""
+            @ load.module("$scriptPath/CompilationError.scala")
+
+            @ val r = res
+            error: Compilation Failed
+            Main.scala:\\d\\+: not found: value res
+            res
+            ^
+
+            """)
+        }
+        'nofile{
+          check.session(s"""
+            @ load.exec("$scriptPath/notHere")
+            error: java.nio.file.NoSuchFileException: $scriptPath/notHere
+            """
+            )
+        }
+        'multiBlockError{
+          check.session(s"""
+            @ load.module("$scriptPath/MultiBlockError.scala")
+
+            @ val r1 = res1
+            r1: Int = 1
+
+            @ val r2 = res2
+            error: Compilation Failed
+            Main.scala:\\d\\+: not found: value res2
+            res2 
+            ^
+            """)
+        }
       }
-      'multiBlockError{
-        check.session(s"""
-          @ load.script("$scriptPath/MultiBlockError.scala")
-
-          @ val r1 = res1
-          r1: Int = 1
-
-          @ val r2 = res2
-          error: Compilation Failed
-          Main.scala:\\d\\+: not found: value res2
-          res2 
-          ^
-          """)
+      'caching{
+        'blocks{
+          'one{
+            val storage = new MemoryStorage
+            val interp = new Interpreter(
```

Comment:
Tests should be DRY too =D this is a non-trivial section of code to initialize the `Interpreter`, pull it out into a def somewhere